nav2_behavior_tree/test/plugins/action/test_planner_selector_node.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2018 Intel Corporation
+// Copyright (c) 2020 Pablo Iñigo Blasco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <set>
+#include <string>
+
+#include "../../test_action_server.hpp"
+#include "behaviortree_cpp_v3/bt_factory.h"
+#include "nav2_behavior_tree/plugins/action/planner_selector_node.hpp"
+#include "nav_msgs/msg/path.hpp"
+#include "std_msgs/msg/string.hpp"
+
+class PlannerSelectorTestFixture : public ::testing::Test
+{
+public:
+  static void SetUpTestCase()
+  {
+    node_ = std::make_shared<rclcpp::Node>("planner_selector_test_fixture");
+    factory_ = std::make_shared<BT::BehaviorTreeFactory>();
+
+    config_ = new BT::NodeConfiguration();
+
+    // Create the blackboard that will be shared by all of the nodes in the tree
+    config_->blackboard = BT::Blackboard::create();
+    // Put items on the blackboard
+    config_->blackboard->set<rclcpp::Node::SharedPtr>("node", node_);
+
+    BT::NodeBuilder builder = [](const std::string & name, const BT::NodeConfiguration & config) {
+        return std::make_unique<nav2_behavior_tree::PlannerSelector>(name, config);
+      };
+
+    factory_->registerBuilder<nav2_behavior_tree::PlannerSelector>("PlannerSelector", builder);
+  }
+
+  static void TearDownTestCase()
+  {
+    delete config_;
+    config_ = nullptr;
+    node_.reset();
+    factory_.reset();
+  }
+
+  void TearDown() override
+  {
+    tree_.reset();
+  }
+
+protected:
+  static rclcpp::Node::SharedPtr node_;
+  static BT::NodeConfiguration * config_;
+  static std::shared_ptr<BT::BehaviorTreeFactory> factory_;
+  static std::shared_ptr<BT::Tree> tree_;
+};
+
+rclcpp::Node::SharedPtr PlannerSelectorTestFixture::node_ = nullptr;
+
+BT::NodeConfiguration * PlannerSelectorTestFixture::config_ = nullptr;
+std::shared_ptr<BT::BehaviorTreeFactory> PlannerSelectorTestFixture::factory_ = nullptr;
+std::shared_ptr<BT::Tree> PlannerSelectorTestFixture::tree_ = nullptr;
+
+TEST_F(PlannerSelectorTestFixture, test_tick)
+{
+  // create tree
+  std::string xml_txt =
+    R"(
+      <root main_tree_to_execute = "MainTree" >
+        <BehaviorTree ID="MainTree">
+          <PlannerSelector selected_planner="{selected_planner}" default_planner="GridBased" topic_name="planner_selector_custom_topic_name"/>
```

Comment:
Another test to quickly show that it publishes to the right default when not specified would be nice to have

(basically just copy paste a simplified version of this test with no `topic_name` in the XML string and check that it still changes)

Reply:
Already added for this third iteration (4th commit)